tests/test_app/tests.py

```diff
@@ -1,12 +1,155 @@
+    serializer.user = user
+
+    with patch('df_auth.settings.api_settings.REQUIRED_AUTH_FIELDS', ['phone_number']):
+        try:
+            assert serializer.is_valid(raise_exception=True)
+        except ValidationError:
+            assert True
+        else:
+            assert False, "ValidationError exception was not raised"
+
+
+def test_token_obtain_serializer_validate_optional_fields():
+    user = User.objects.create_user(email='test@test.com', password='12345')
+    payload = {'email': 'test@test.com', 'password': '12345'}
+    serializer = TokenObtainSerializer(data=payload)
+    serializer.user = user
+
+    with patch('df_auth.settings.api_settings.REQUIRED_AUTH_FIELDS', ['email', 'password']):
+        with patch('df_auth.settings.api_settings.OPTIONAL_AUTH_FIELDS', ['phone_number']):
+            assert serializer.is_valid(raise_exception=True) is True
+
+
+def test_otp_obtain_serializer_get_fields():
+    serializer = OTPObtainSerializer()
+    expected_fields = ['email', 'phone_number', 'password', 'otp']
+    fields = serializer.get_fields()
+    assert len(fields.keys()) == len(expected_fields)
+    assert all([field in fields.keys() for field in expected_fields])
+
+
+def test_otp_obtain_serializer_validate():
+    factory = RequestFactory()
+    user = User.objects.create_user(email='test@test.com', password='12345')
+    payload = {'email': 'test@test.com', 'password': '12345'}
+    request = factory.get('/')
+    request.user = user
+    serializer = OTPObtainSerializer(data=payload, context={'request': request})
+    serializer.user = user
+    assert serializer.is_valid(raise_exception=True) is True
+
+
+def test_sign_up_serializer_validate():
+    factory = RequestFactory()
+    payload = {'email': 'test@test.com', 'password': '12345', 'first_name': 'Test', 'last_name': 'Tester'}
+    request = factory.get('/')
+    serializer = SignupSerializer(data=payload, context={'request': request})
+    assert serializer.is_valid(raise_exception=True) is True
+
+
+def test_invite_serializer_validate():
+    factory = RequestFactory()
+    user = User.objects.create_user(email='test@test.com', password='12345')
+    payload = {'email': 'test@test.com'}
+    request = factory.get('/')
+    request.user = user
+    serializer = InviteSerializer(data=payload, context={'request': request})
+    serializer.user = user
+    assert serializer.is_valid(raise_exception=True) is True
+
+
+class TestAuthBackendSerializerMixin:
+    def setup_method(self):
+        self.serializer = AuthBackendSerializerMixin()
+
+    def test_get_fields(self):
+        fields = self.serializer.get_fields()
+        assert isinstance(fields, dict)
+
+    def test_validate_success(self):
+        attrs = {
+            'username': 'test_user',
+            'password': 'test_password',
+        }
+        self.serializer.backend_method_name = ""
+        with pytest.raises(AuthenticationFailed):
+            self.serializer.validate(attrs)
```

Comment:
This doesn't look like a 'success'. Did you mean to say 'test_validate_invalid_backend'?